kratos/solving_strategies/strategies/explicit_solving_strategy_runge_kutta_4.h

```diff
@@ -0,0 +1,414 @@
+        : BaseType(rModelPart, ThisParameters)
+    {
+    }
+
+    /**
+     * @brief Default constructor.
+     * @param rModelPart The model part to be computed
+     * @param pExplicitBuilderAndSolver The pointer to the explicit builder and solver
+     * @param MoveMeshFlag The flag to set if the mesh is moved or not
+     */
+    explicit ExplicitSolvingStrategyRungeKutta4(
+        ModelPart &rModelPart,
+        typename ExplicitBuilderAndSolverType::Pointer pExplicitBuilderAndSolver,
+        bool MoveMeshFlag = false,
+        int RebuildLevel = 0)
+        : BaseType(rModelPart, pExplicitBuilderAndSolver, MoveMeshFlag, RebuildLevel)
+    {
+    }
+
+    /**
+     * @brief Default constructor.
+     * @param rModelPart The model part to be computed
+     * @param MoveMeshFlag The flag to set if the mesh is moved or not
+     */
+    explicit ExplicitSolvingStrategyRungeKutta4(
+        ModelPart &rModelPart,
+        bool MoveMeshFlag = false,
+        int RebuildLevel = 0)
+        : BaseType(rModelPart, MoveMeshFlag, RebuildLevel)
+    {
+    }
+
+    /** Copy constructor.
+     */
+    ExplicitSolvingStrategyRungeKutta4(const ExplicitSolvingStrategyRungeKutta4 &Other) = delete;
+
+    /** Destructor.
+     */
+    virtual ~ExplicitSolvingStrategyRungeKutta4() = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "ExplicitSolvingStrategyRungeKutta4";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream &rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream &rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    void SolveWithLumpedMassMatrix() override
+    {
+        // Get the required data from the explicit builder and solver
+        const auto p_explicit_bs = BaseType::pGetExplicitBuilderAndSolver();
+        auto& r_dof_set = p_explicit_bs->GetDofSet();
+        const unsigned int dof_size = p_explicit_bs->GetEquationSystemSize();
+        const auto& r_lumped_mass_vector = p_explicit_bs->GetLumpedMassMatrixVector();
+
+        // Set the auxiliary RK vectors
+        LocalSystemVectorType u_n(dof_size); // TODO: THIS IS INEFICCIENT. CREATE A UNORDERED_SET WITH THE IDOF AND VALUE AS ENTRIES. THIS HAS TO BE OPTIONAL
+        LocalSystemVectorType rk_k1(dof_size);
+        LocalSystemVectorType rk_k2(dof_size);
+        LocalSystemVectorType rk_k3(dof_size);
+        LocalSystemVectorType rk_k4(dof_size);
```

Comment:
this comment is related to the const member variables `mA21`... posted by @loumalouomega (https://github.com/KratosMultiphysics/Kratos/pull/7138/files#r445705855)

Since this is a particular RK4 method (coefficients a_31 = a_41 = a_42 = 0), I guess you can store the contributions directly to the final stage and avoid the intermediate `rk_k2`, `rk_k3` and `rk_k4` variables. It could help to reduce memory consumption

Reply:
I think I didn't get your point at all. You mean directly add the corresponding contributions to the nodal database?

Reply:
Adding directly to `u_n`. In that case, the `PerformRungeKuttaIntermediateSubStep` method will need to know also `mB1`, `mB2`...  and possibly `r_u_old`

I don't remember the details of implementation, maybe there is a performance reason to keep all the intermediate variables (accessing the system vector `u_n` fewer times)

Reply:
OK. I agree that for this particular RK4 scheme it can be alternatively implemented as you point out. What I'm not sure at all is which option will be more efficient, however I think that your approach is a little bit cumbersome as it will mix total values and increments within the `u_n` vector...